### PR TITLE
Hack up make_fx to natively support varargs

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83210

This is kind of nasty but it works.  I attempted to fix FX
first but the inspect logic is impenetrable.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by Chillee** [comment]
I think at some point I tried the same thing (changing FX inspect logic lol), and also gave up.

**Review by Chillee — APPROVED** [review]
LGTM

File: torch/fx/experimental/proxy_tensor.py [hunk 1]
```diff
@@ -27,6 +28,12 @@
 CURRENT_DECOMPOSITION_TABLE: Dict[torch._ops.OpOverload, Callable] = {}
 
 
+def fake_signature(fn, nargs):
+    """FX gets confused by varargs, de-confuse it"""
+    argnames = ",".join(f"arg{i}" for i in range(nargs))
+    return eval(f"lambda {argnames}: fn({argnames})", {"fn": fn})
```
Comment (albanD) [inline torch/fx/experimental/proxy_tensor.py:34]: make_fx doesn't support kwargs right now?

**Review by albanD — APPROVED** [review]

File: torch/fx/experimental/proxy_tensor.py (hunk 1 above)
Comment (ezyang) [reply torch/fx/experimental/proxy_tensor.py:34]: it doesn't. Actually @IvanYashchuk added another wrapper to make kwargs work with pytree...

**Pull request closed without merging**